lab-2/virtual-assistant/riva_local/asr/asr.py

```diff
@@ -150,43 +148,26 @@ def main_asr(self):
         # for a list of supported languages.
         self.start()
 
-        config = rasr.RecognitionConfig(
-            encoding=ra.AudioEncoding.LINEAR_PCM,
-            sample_rate_hertz=self.sampling_rate,
-            language_code=self.language_code,
-            max_alternatives=1,
-            enable_automatic_punctuation=self.enable_automatic_punctuation,
-            verbatim_transcripts=True
-        )
-        streaming_config = rasr.StreamingRecognitionConfig(
-            config=config,
-            interim_results=self.stream_interim_results)
+        config = riva.client.RecognitionConfig()
+        config.sample_rate_hertz = self.sampling_rate
+        config.language_code = self.language_code
+        config.max_alternatives = 1    
+        config.enable_automatic_punctuation = self.enable_automatic_punctuation
+        config.verbatim_transcripts = True
+        config.audio_channel_count = 1 
+
+        streaming_config = riva.client.StreamingRecognitionConfig(config=config, interim_results=True)
 
         if self.verbose:
             print("[Riva ASR] Starting Background ASR process")
 
         self.request_generator = self.build_request_generator()
 
-        requests = (rasr.StreamingRecognizeRequest(audio_content=content)
-                    for content in self.request_generator)
-
-        def build_generator(cfg, gen):
-            # First message sent must include only a StreamingRecognitionConfig.
-            yield rasr.StreamingRecognizeRequest(streaming_config=cfg)
-
-            # Subsequent messages sent in the stream must contain only raw bytes
-            # of the audio to be recognized.
-            for x in gen:
-                yield x
-            yield cfg
-
-        streaming_recognize_input = build_generator(streaming_config, requests)
-
         if self.verbose:
             print("[Riva ASR] StreamingRecognize Start")
 
         # <------------ EXERCISE: Fill in the line of code below ------------->
-        # responses = self.asr_client.xx ?
+        # responses = self.riva_asr.streaming_response_generator(xx) ?
 
         # Now, put the transcription responses to use.
         self.listen_print_loop(responses)
```

Comment:
I think the VA web-app uses the transcript printed by this method (on stdout) for displaying in the chatbot interface and to send the final transcript (not the intermediate streaming one) to the NER model.

The old method `listen_print_loop(responses)` throws error with the new Riva client and hence I've modified this line to use [print_streaming](https://github.com/nvidia-riva/python-clients/blob/main/riva/client/asr.py#L105). 

After using this method, the final transcripts are not getting sent to the NER model. It's only when I type the query in the chatbot, the NER and the stateDM methods are called. In addition, the chatbot interface is also not updated with the intermediate, nor the final ASR transcript.